pkg/v1/providers/ytt/02_addons/management-packages/add_management_package_repo.yaml

```diff
@@ -0,0 +1,49 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:yaml", "yaml")
+#@ load("/lib/helpers.star", "get_default_tkg_bom_data", "get_image_repo_for_component")
+
+#@ def management_package_repo():
+#@ management_package_repo_bundle = get_default_tkg_bom_data().components["tanzu-framework-management-packages"][0].images.tanzuFrameworkManagementPackageRepositoryImage
+---
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageRepository
+metadata:
+  name: tanzu-management
+  namespace: tkg-system
+spec:
+  fetch:
+    imgpkgBundle:
+      image: #@ "{}/{}:{}".format(get_image_repo_for_component(management_package_repo_bundle), management_package_repo_bundle.imagePath, management_package_repo_bundle.tag)
+#@ end
+
+#@ if data.values.TKG_CLUSTER_ROLE != "workload":
+#@ if data.values.DISABLE_CRS_FOR_ADDON_TYPE and "packages/management-package-repo" in data.values.DISABLE_CRS_FOR_ADDON_TYPE:
+--- #@ management_package_repo()
+#@ else:
+---
+apiVersion: addons.cluster.x-k8s.io/v1alpha3
+kind: ClusterResourceSet
+metadata:
+  name: #@ "{}-management-package-repository".format(data.values.CLUSTER_NAME)
+  labels:
+    cluster.x-k8s.io/cluster-name: #@ data.values.CLUSTER_NAME
+  annotations:
+    tkg.tanzu.vmware.com/addon-type: "packages/management-package-repo"
```

Comment:
Are we going to apply multiple packageInstalls from managment-package-repo in the future? If so, I think it's better to give each component it's subfolder. Otherwise, the deployment of some package can get more complex once they require `values.yaml` to be specified, and things will mix together.

Reply:
great point.. i move it to a sub folder